python/ray/tests/test_advanced_2.py

```diff
@@ -745,6 +748,73 @@ def f():
     wait_for_pid_to_exit(pid1)
 
 
+def test_worker_leaked_when_drain(ray_start_regular):
+    @ray.remote
+    def hello():
+        return "hello"
+
+    # Warnup cluster
+    assert "hello" == ray.get(hello.remote())
+
+    driver_template = """
+import ray
+import os
+import ray
+import numpy as np
+import time
+
+ray.init(address="{address}", namespace="test")
+
+@ray.remote
+class KvStore:
+    def __init__(self):
+        self.__data = dict()
+
+    def put(self, k, v):
+        self.__data[k] = v
+        return True
+
+    def get(self, k):
+        return self.__data[k]
+
+@ray.remote
+def normal_task(large):
+    try:
+        kvstore = ray.get_actor("kvstore", namespace="test")
+    except Exception as e:
+        kvstore = KvStore.options(name="kvstore", lifetime="detached").remote()
+    ray.get(kvstore.put.remote("pid", os.getpid()))
+    time.sleep(60 * 60)
+    return "normaltask"
+
+large = ray.put(np.zeros(100 * 2**10, dtype=np.int8))
+obj = normal_task.remote(large)
+print(ray.get(obj))
+"""
+    driver_script = driver_template.format(
+        address=ray_start_regular["redis_address"])
+
+    driver_proc = run_string_as_driver_nonblocking(driver_script)
+
+    try:
+        driver_proc.wait(10)
+    except Exception as e:
+        print(e)
+
+    kvstore = ray.get_actor("kvstore", namespace="test")
+    normal_task_pid = ray.get(kvstore.get.remote("pid"))
```

Comment:
Could you add more comments in the test case? I don't quite understand what scenario it tries to reproduce.